### PR TITLE
feat(pie-docs): DSW-1091 add component overview pages D to I

### PR DESCRIPTION
**Pull request by FayeCarter**
## Describe your changes (can list changeset entries if preferable)
add component overview pages D to I

## Author Checklist (complete before requesting a review)
- [ ] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests
- [ ] If it is a `PIE Docs` change, I have reviewed the Docs site preview
- [ ] If it is a component change, I have reviewed the Storybook preview
- [ ] If there are visual test updates, I have reviewed them properly before approving

### Reviewer 1
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### Reviewer 2
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### CODE REVIEW

File: apps/pie-docs/src/assets/styles/components/_nav.scss [hunk 1]
```diff
@@ -83,7 +89,18 @@ $nav-item-left-margin: calc($nav-category-gap + $nav-category-icon-width);
     }
 }
 
+.c-nav-subCategory {
+    padding-block: 10px;
+    padding-inline: calc(f.spacing(d) + $nav-item-left-margin) f.spacing(d);
+    margin-block: 0 2px;
+
+    + .c-nav-list {
+        margin-left: f.spacing(d);
```
Comment (raoufswe) [inline apps/pie-docs/src/assets/styles/components/_nav.scss:98]:
```suggestion
        margin-inline-start: f.spacing(d);
```

Comment (FayeCarter) [reply apps/pie-docs/src/assets/styles/components/_nav.scss:98]: Ah damn, I thought I'd got them all, Thanks :)

File: apps/pie-docs/src/_includes/nav.njk [hunk 2]
```diff
@@ -16,6 +16,31 @@
...
+        {{ title }}
+        {{ { name: 'chevron-up', attrs: { fill: chevronFill, height: '16', width: '16', class:'c-nav-toggle c-nav-toggle--close', 'aria-hidden': 'true' } } | pieIconsSvg | safe }}
+        {{ { name: 'chevron-down', attrs: { fill: chevronFill, height: '16', width: '16', class:'c-nav-toggle c-nav-toggle--open', 'aria-hidden': 'true' } } | pieIconsSvg | safe }}
+    </summary>
+{% endmacro %}
+
+{% macro listItem(title, url, pageUrl, isActive) %}
+    <li>
+        <a
+            href="{{ url | url}}"
```
Comment (raoufswe) [inline apps/pie-docs/src/_includes/nav.njk:34]: isn't this doing the same thing 👀

Comment (raoufswe) [reply apps/pie-docs/src/_includes/nav.njk:34]: so the first url will be different from the second url?

I see it was `href="{{ subpage.url | url}}"` before

**Review by raoufswe — APPROVED** [review]

**Review by LTurns — APPROVED** [review]

**Pull request merged**